frontend/src/components/worksheets/WorksheetItemList.js

```diff
@@ -251,7 +257,22 @@ class WorksheetItemList extends React.Component {
                     );
                 }.bind(this),
             );
-            items_display = worksheet_items;
+            items_display = (
+                <>
+                    {worksheet_items}
+                    <NewUpload
+                        key={this.state.newUploadKey}
```

Comment:
While changing the key is a very smart way to deal with the rerendering issue, I wonder if there's a more readable way to deal with issues like this.

Reply:
I believe the other way you linked to recommends setting `el.target.value` to an empty string. It seems like a hack and not documented behavior (https://stackoverflow.com/questions/20549241/how-to-reset-input-type-file) so I think it's better to change the key just to be safe, so that the entire elemnt re-renders.